device/iot-device-samples/send-event/src/main/java/samples/com/microsoft/azure/sdk/iot/InputParameters.java

```diff
@@ -0,0 +1,139 @@
+                                .hasArg()
+                                .desc("Path to certificate to enable one-way authentication over ssl (not necessary when connecting directly to Iot Hub, but required if connecting to an Edge device using a non public root CA certificate)")
+                                .build()
+                );
+
+        String cmdLine = APPEXE;
+        try {
+            String jarPath = Options.class
+                    .getProtectionDomain()
+                    .getCodeSource()
+                    .getLocation()
+                    .toURI()
+                    .getPath();
+            String jarName = jarPath.substring(jarPath.lastIndexOf("/") + 1);
+            cmdLine += jarName;
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+            cmdLine += "<sample.jar>";
+        }
+
+        // parse command line
+        CommandLineParser parser = new DefaultParser();
+        HelpFormatter formatter = new HelpFormatter();
+
+        try {
+            // parse the command line arguments
+            cmd = parser.parse(options, args);
+
+            // -h option
+            if(cmd.hasOption("h")){
+                formatter.printHelp(cmdLine, "\nHelp:\n\n", options, FOOTER, true);
+                System.exit(0);
+            }
+
+            // -c and -r option
+            if(!cmd.hasOption("c") || !cmd.hasOption("r")){
+                formatter.printHelp(cmdLine, "\nError: ConnectionString and NumberOfRequests are required as arguments\n\nHelp:\n", options, FOOTER, true);
+                System.exit(0);
+            }else{
+                if(cmd.getOptionValue("c") == null || cmd.getOptionValue("c").trim().isEmpty()){
+                    formatter.printHelp(cmdLine, "\nError: ConnectionString is empty\n\nHelp:\n", options, FOOTER, true);
+                    System.exit(0);
+                }
+
+                if(cmd.getOptionValue("r") == null || cmd.getOptionValue("r").trim().isEmpty()){
+                    formatter.printHelp(cmdLine, "\nError: NumberOfRequests is empty\n\nHelp:\n", options, FOOTER, true);
+                    System.exit(0);
+                }
+            }
+
+        } catch (ParseException e) {
+            //wrong parameters
+            formatter.printHelp(cmdLine, "\nError: "+e.getMessage()+"\n\nHelp:\n", options, FOOTER, true);
+            System.exit(0);
+        }
+    }
+
+    /**
+     * get connection string argument from command line
+     * @return string value
+     */
+    public String getConnectionString() { return cmd.getOptionValue("c"); }
+
+
+    /**
+     * get number of requests argument from command line
+     * @return string value
+     */
+    public String getNumberOfRequests() { return cmd.getOptionValue("r"); }
+
+
+    /**
+     * get protocol argument from command line
+     * @return string value
+     */
+    public String getProtocol() { return cmd.getOptionValue("p", PROTOCOL); }
```

Comment:
Can this be the actual protocol enum value instead of a string? or a command line level Enum value? So we don't have to further parse it in the samples itself. just curious.

Reply:
> agree, will change this.